daemon/policy.go

```diff
@@ -93,11 +93,53 @@ func (d *Daemon) TriggerPolicyUpdates(added []policy.NumericIdentity) {
 			if policyChanges {
 				ep.Regenerate(d)
 			}
+			// Only update whether policy is enabled after rules are added successfully for this endpoint.
+			d.updatePolicyEnforcement(ep)
```

Comment:
When we enable policy enforcement on an endpoint, we regenerate the endpoint which means in the above loop, we regenerate it twice if `TriggerPolicyUpdates` already did.